### PR TITLE
Query by id and type

### PR DESCRIPTION
**Pull request by MinervaMint**
Added route `/recommendations/<id>/type/<typeid>` for querying recommendations by product id and recommendation type;
Added classmethod `find_by_id_type` in model.py;
Added test cases for querying active recommendations, and querying by id and type.

### CODE REVIEW

File: service/model.py [hunk 1]
```diff
@@ -154,10 +155,16 @@ def find_or_404(cls, by_id):
...
     @classmethod
     def find_by_id_status(cls, by_id: int, by_status=True):
-        """ Find active recommendations of a product [id] """
+        """ Find [status: active/inactive] recommendations of a [product: id] """
         cls.logger.info("Processing lookup for id %s with status %s", by_id, by_status)
         return cls.query.filter(cls.id==by_id, cls.status==by_status)
 
+    @classmethod
+    def find_by_id_type(cls, by_id: int, by_type: int):
+        """ Find recommendations of a [product: id] with [type: typeid] """
```
Comment (sushanthsamala) [inline service/model.py:169]: Could you please raise TypeErrors when by_id or by_type are not integers and raise a DataValidationError when by_type is not in [1, 2, 3]. You can take a look at find_recommendation() and de-serialize function for reference

File: service/service.py [hunk 2]
```diff
@@ -120,6 +120,28 @@ def get_active_related_products(id):
...
+######################################################################
+# QUERY RECOMMENDATIONS BY ID AND TYPE
+######################################################################
+@app.route('/recommendations/<int:id>/type/<int:typeid>', methods=['GET'])
+def get_related_products_with_type(id, typeid):
+    app.logger.info("Query type: %s recommendations for id: %s", typeid, id)
+    recommendations = Recommendation.find_by_id_type(id, typeid)
+
+    if not recommendations:
+        raise NotFound("Type {} recommendations for product {} not found".format(typeid, id))
```
Comment (sushanthsamala) [inline service/service.py:148]: This line was not covered in unit tests, could you please add a test case to capture this line

**Review by sushanthsamala — CHANGES_REQUESTED** [review]

File: service/service.py (hunk 2 above)
Comment (sushanthsamala) [reply service/service.py:148]: You can take a look at line 81 in tests/test_recommendation.py for reference

File: service/model.py (hunk 1 above)
Comment (MinervaMint) [reply service/model.py:169]: Oh okay sure. Thank you for reminding!

**Comment by MinervaMint** [comment]
Modified test cases;
Added type check for functions in model.py.

File: service/model.py [hunk 3]
```diff
@@ -154,10 +155,26 @@ def find_or_404(cls, by_id):
...
+
         cls.logger.info("Processing lookup for id %s with status %s", by_id, by_status)
         return cls.query.filter(cls.id==by_id, cls.status==by_status)
 
+    @classmethod
+    def find_by_id_type(cls, by_id: int, by_type: int):
+        """ Find recommendations of a [product: id] with [type: typeid] """
+        if not by_id or not isinstance(by_id, int):
+            raise TypeError("by_id is not of type int")
+        if not 1 <= by_type <= 3:
```
Comment (sushanthsamala) [inline service/model.py:174]: Awesome, great work. One last thing, could you add integer type check for "by_type" also, similar to what you did for by_id?

Comment (MinervaMint) [reply service/model.py:174]: Yes sure I just added the type check. Sorry for the delay.

**Review by sushanthsamala — APPROVED** [review]
Make sure to merge latest master before merging your branch.

**Pull request merged**